modules/images/webp-uploads/load.php

```diff
@@ -469,11 +470,25 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
 
 	// Replace the full size image if present.
 	if ( isset( $metadata['sources'][ $target_mime ]['file'] ) ) {
+		// Initially set the target mime as the replacement source.
+		$replacement_source = $metadata['sources'][ $target_mime ]['file'];
+
+		// Check for the smaller image file.
+		if (
+			$prefer_smaller_image_file &&
+			! empty( $metadata['sources'][ $target_mime ]['filesize'] ) &&
+			! empty( $metadata['sources'][ $original_mime ]['filesize'] ) &&
```

Comment:
@adamsilverstein That's a good point, I think eventually that would make sense. It would require a bit more complex reworking though indeed, for example right now we only have a single `$target_mime`, and then there will need to be a list, all of which we would need to go through.

+1 to doing it in a separate issue.

Reply:
I'm taking the "look thru every mime type for the smallest" approach for the (follow up) core patch, basically go thru all sources to find the smallest and use that:
```php
foreach( $metadata['sources'] as $source ) {
		if ( empty( $source['filesize'] ) ) {
			continue;
		}
		if ( $source['filesize'] < $target_file['filesize'] ) {
			$target_file = $source;
		}
	}
```